File: 53-maximum-subarray/maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int curr = 0;
        int max= Integer.MIN_VALUE;

        for(int temp : nums){
            curr += temp;

            max = Math.max(curr, max);

            if(curr < 0){
                curr = 0;
            }
        }
        return max;
    }
}

File: 53-maximum-subarray/maximum-subarray.py
import sys
class Solution(object):
    def maxSubArray(self, nums):
        Max = -99999
        Sum = 0

        for i in range (len(nums)) :
            Sum += nums[i]
            Max = max(Max,Sum)
            if Sum < 0 : Sum = 0

        return Max
        
        